### DISCUSSION TITLE
Cmake Build Engine Error

### DISCUSSION THREAD

**Discussion by RazaelDAbyss** [o3de/o3de#5850] [Help Wanted] [answered]
I get the following error message at the "Build the Engine" step:
cmake -E tar: warning: unknown warning
CMake Error: Problem with archive_write_finish_entry(): File size could not be restored
CMake Error: Problem extracting tar: C:/o3de/o3de-packages/downloaded_packages/AWSNativeSDK-1.7.167-rev4-windows.tar.xz
-- Using package C:/o3de/o3de-packages/packages/AWSNativeSDK-1.7.167-rev4-windows
-- Downloading package into C:/o3de/o3de-packages/packages/assimp-5.0.1-rev11-multiplatform
CMake Error at cmake/3rdPartyPackages.cmake:287 (message):
  ly_package: - Unable to get package assimp-5.0.1-rev11-multiplatform from
  any download server.  Enable LY_PACKAGE_DEBUG to debug.

  Error from server https://d3t6xeg4fgfoum.cloudfront.net - 1 - Downloaded
  successfully, but the file hash did not match expected hash!
Call Stack (most recent call first):
  cmake/3rdPartyPackages.cmake:499 (ly_package_internal_download_package)
  cmake/3rdPartyPackages.cmake:589 (ly_force_download_package)
  cmake/3rdPartyPackages.cmake:690 (ly_enable_package)
  cmake/LYWrappers.cmake:483 (ly_download_associated_package)
  cmake/LYWrappers.cmake:234 (ly_parse_third_party_dependencies)
  Code/Tools/SceneAPI/SDKWrapper/CMakeLists.txt:15 (ly_add_target)

-- Configuring incomplete, errors occurred!

The command line instruction is:
cmake -B build/windows_vs2019 -G "Visual Studio 16" -DLY_3RDPARTY_PATH=C:\o3de\o3de-packages -DLY_UNITY_BUILD=ON -DLY_PROJECTS=AutomatedTesting

Any idea what is causing this?  I've tried several times and get the same error message.

**Comment by nick-l-o3de** [comment, marked as answer]
Hmm, not sure why this is happening.  I grabbed the same file myself off the network and checked its hash code and it matches:

```
$ curl https://d3t6xeg4fgfoum.cloudfront.net/AWSNativeSDK-1.7.167-rev4-windows.tar.xz -o AWSNativeSDK-1.7.167-rev4-windows.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  138M  100  138M    0     0  10.6M      0  0:00:13  0:00:13 --:--:-- 10.5M

$ shasum -a 256 AWSNativeSDK-1.7.167-rev4-windows.tar.xz 
a900e80f7259e43aed5c847afee2599ada37f29db70505481397675bcbb6c76c  AWSNativeSDK-1.7.167-rev4-windows.tar.xz

$ cat cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake | grep -i awsnativesdk
ly_associate_package(PACKAGE_NAME AWSNativeSDK-1.7.167-rev4-windows                     TARGETS AWSNativeSDK                PACKAGE_HASH a900e80f7259e43aed5c847afee2599ada37f29db70505481397675bcbb6c76c)
```

so the expected package sha256 hash vs the actual hash is
```
 a900e80f7259e43aed5c847afee2599ada37f29db70505481397675bcbb6c76c
 a900e80f7259e43aed5c847afee2599ada37f29db70505481397675bcbb6c76c
```

The file on the website at https://d3t6xeg4fgfoum.cloudfront.net/AWSNativeSDK-1.7.167-rev4-windows.tar.xz  definitely has the right data in it but somehow by the time it gets to your PC its corrupted somehow during download?

Anything special happening on that PC or configured on that PC?  You could try downloading the file yourself using the above URL and put it in your user home folder (ie, c:\users\yourname\.o3de\downloaded_packages and seeing if it accepts it.  If you have 7z you can also check its hash after downloading.

something is corrupting that file during download onto your PC though, since its clearly OK on the site.

**Reply by RazaelDAbyss** [reply]
Sorry for the post.  I figured out I had a typo in path name.
